ENH: Clean up scripts

ZLLentz opened the pull request:
A step toward addressing #115 

Fixes some minor things and tries to make the repo less "magical", e.g. avoid issues with trying to guess exactly what we want to be testing and removing more of the "manual" steps.

- adds pymongo for the happi tests (and in general)
- stop trying to guess which tests we want to run, require a list of repos to clone
- clean up the test setup implementation
- automatically update critical pins to latest on staging a new environment

Lots of packages are broken right now so I can't really do a release, I want to merge this and then go around fixing all the packages separately. A lot of stuff was depending on the old location of `SignalRO`

klauer commented on scripts/test_setup.py:
```diff
@@ -1,23 +1,62 @@
+    repo_file = Path(repo_file)
+
+    with repo_file.open('r') as fd:
+        repos = fd.read().strip().split('\n')
```
Not necessary to change, but a convenience method I've been moving to recently:
```suggestion
        repos = fd.read().strip().splitlines()
```

ZLLentz replied:
That's a nifty one

klauer commented on scripts/test_setup.py:
```diff
@@ -32,36 +71,15 @@ def pushd(new_dir):
+    else:
+        tags = None
+
+    setup_tests(repo_file, tags=tags)
```
Also not required, but maybe `setup_tests` becomes `setup_test` and you loop through the repos here?
```
for repo in read_repo_file_list:
   setup_test(repo, tag=tags.get(repo) if tags else None)
```

ZLLentz replied:
I agree with this one, I'll revise later

ZLLentz commented:
Implemented Ken's suggestions, added an error message for a situation that hit me this morning while refactoring, and isolated main into a function instead of having it hang out in the if name block. Merging and moving on to the other issues on this repo.
